plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/IngestModule.java

```diff
@@ -20,31 +20,11 @@
 package org.elasticsearch.plugin.ingest;
 
 import org.elasticsearch.common.inject.AbstractModule;
-import org.elasticsearch.common.inject.multibindings.MapBinder;
-import org.elasticsearch.ingest.processor.append.AppendProcessor;
-import org.elasticsearch.ingest.processor.convert.ConvertProcessor;
-import org.elasticsearch.ingest.processor.date.DateProcessor;
-import org.elasticsearch.ingest.processor.fail.FailProcessor;
-import org.elasticsearch.ingest.processor.geoip.GeoIpProcessor;
-import org.elasticsearch.ingest.processor.grok.GrokProcessor;
-import org.elasticsearch.ingest.processor.gsub.GsubProcessor;
-import org.elasticsearch.ingest.processor.join.JoinProcessor;
-import org.elasticsearch.ingest.processor.lowercase.LowercaseProcessor;
-import org.elasticsearch.ingest.processor.remove.RemoveProcessor;
-import org.elasticsearch.ingest.processor.rename.RenameProcessor;
-import org.elasticsearch.ingest.processor.set.SetProcessor;
-import org.elasticsearch.ingest.processor.split.SplitProcessor;
-import org.elasticsearch.ingest.processor.trim.TrimProcessor;
-import org.elasticsearch.ingest.processor.uppercase.UppercaseProcessor;
 import org.elasticsearch.plugin.ingest.rest.IngestRestFilter;
 
-import java.util.HashMap;
-import java.util.Map;
-
 public class IngestModule extends AbstractModule {
 
     private final boolean ingestEnabled;
-    private final Map<String, ProcessorFactoryProvider> processorFactoryProviders = new HashMap<>();
 
     public IngestModule(boolean ingestEnabled) {
         this.ingestEnabled = ingestEnabled;
```

Comment:
Ok, so:
* Processors that have no external dependencies and don't require extra security permissions can be part of core with the rest of the ingest infrastructure. There is no reason to isolate this processor implementation. But tests for the ingest infrastructure should be processor agnostic and rely on mocks.
* Processors that should be available by default but require additional dependencies or security permissions should  be part of the ingest-processor module.
* Processors that can't be part of the ingest-processor module because of technical reasons or we don't them to be shipped by default should become separate plugins.